appium-cookies/lib/cookies.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * derived from jQuery Cookie Plugin v1.4.1
+ * https://github.com/carhartl/jquery-cookies
+ */
+
+ //needed to communicate/translate between JSONWire cookies and regular JS cookies 
+
+import _ from 'lodash';
+
+export default class Cookies {
+
+  constructor (cookieString) {
+    this.cookieString = cookieString;
+  }
+
+  //parses the value if needed and converts the value if a converter is provided
+  convertCookie(value, converter) {
+    let parsedValue;
+
+    if (value.indexOf(`"`) === 0) {
+      value = value.slice(1, -1).replace(/\\"/g, `"`).replace(/\\\\/g, `\\`);
+    }
+    try {
+      parsedValue = decodeURIComponent(value.replace(/\+/g, ` `));
+    } catch (e) {}
+    
+    return _.isFunction(converter) ? converter(parsedValue) : parsedValue;
+  }
+
+  // takes arguments given and creates a JS Cookie 
+  writeCookie(key, value, options) { 
+    if (arguments.length > 1 && !_.isFunction(value)) {
+      options = _.extend({}, options);
+
+      let ret = [
+        encodeURIComponent(key), `=`, value, 
+        options.expires ? `; expires=${options.expires}` : ``,
+        options.path    ? `; path=${options.path}`       : ``,
+        options.domain  ? `; domain=${options.domain}`   : ``, 
+        options.secure  ? `; secure` : ``
+      ].join(``);
+      
+      return ret;
+    }
+  }
+
+  //takes the cookieString given in the constructor and creates a JSONWire cookie out of it
+  readCookie(key, converter) {
+    let result = key ? undefined : {};
+    let cookies = this.cookieString ? this.cookieString.split(`; `) : [];
+
+    for (let cookie of cookies) {
+      let parts = cookie.split(`=`);
+      let val = parts[1];
+      let name = decodeURIComponent(parts.shift());
+
+      if (key && key === name) {
+        result = this.convertCookie(val, converter);
+        break;
+      }
+
+      if (!key && (cookie = this.convertCookie(cookie)) !== undefined) {
```

Comment:
this code might have been this way before, but we have a convention of not comparing directly with `undefined`. first, a question. why are we redefining `cookie` here? it doesn't seem that we use it so why assign a new value to the same name?

and a style comment. we try not to ever do comparisons directly with `undefined` (because it can be given another value; that's a JS gotcha). instead, we check that the type of something is undefined. there's a lodash helper for this `_.isUndefined`. So I would rewrite this line as (assuming there's no reason we're reassigning `cookie`):

```js
if (!key && !_.isUndefined(this.convertCookie(cookie))) {
```